runelite-client/src/main/java/net/runelite/client/plugins/questlist/QuestListPlugin.java

```diff
@@ -0,0 +1,401 @@
+	}
+
+	private void openSearch()
+	{
+		updateFilter("");
+		questSearchButton.setAction(1, MENU_CLOSE);
+		questSearchButton.setOnOpListener((JavaScriptCallback) this::menuClose);
+		searchInput = chatboxPanelManager.openTextInput("Search quest list")
+			.onChanged(s -> clientThread.invokeLater(() -> updateFilter(s)))
+			.onClose(() ->
+			{
+				clientThread.invokeLater(() -> updateFilter(""));
+				questSearchButton.setOnOpListener((JavaScriptCallback) this::menuOpen);
+				questSearchButton.setAction(1, MENU_OPEN);
+			})
+			.build();
+	}
+
+	private void updateFilter()
+	{
+		String filter = "";
+		if (isChatboxOpen())
+		{
+			filter = searchInput.getValue();
+		}
+
+		updateFilter(filter);
+	}
+
+	private void updateFilter(String filter)
+	{
+		filter = filter.toLowerCase();
+		final Widget container = client.getWidget(WidgetInfo.QUESTLIST_CONTAINER);
+
+		final Widget freeList = client.getWidget(QuestContainer.FREE_QUESTS.widgetInfo);
+		final Widget memberList = client.getWidget(QuestContainer.MEMBER_QUESTS.widgetInfo);
+		final Widget miniList = client.getWidget(QuestContainer.MINI_QUESTS.widgetInfo);
+
+		if (container == null || freeList == null || memberList == null || miniList == null)
+		{
+			return;
+		}
+
+		updateList(QuestContainer.FREE_QUESTS, filter);
+		updateList(QuestContainer.MEMBER_QUESTS, filter);
+		updateList(QuestContainer.MINI_QUESTS, filter);
+
+		memberList.setOriginalY(freeList.getOriginalY() + freeList.getOriginalHeight() + ENTRY_PADDING);
+		miniList.setOriginalY(memberList.getOriginalY() + memberList.getOriginalHeight() + ENTRY_PADDING);
+
+		freeList.revalidateScroll();
+		memberList.revalidateScroll();
+		miniList.revalidateScroll();
+
+		int y = miniList.getRelativeY() + miniList.getHeight();
+
+		int newHeight = 0;
+		if (container.getScrollHeight() > 0)
+		{
+			newHeight = (container.getScrollY() * y) / container.getScrollHeight();
+		}
+
+		container.setScrollHeight(y);
+		container.revalidateScroll();
+
+		client.runScript(
+			ScriptID.UPDATE_SCROLLBAR,
+			WidgetInfo.QUESTLIST_SCROLLBAR.getId(),
+			WidgetInfo.QUESTLIST_CONTAINER.getId(),
+			newHeight
+		);
+	}
+
+	private void updateList(QuestContainer questContainer, String filter)
+	{
+		Widget list = client.getWidget(questContainer.widgetInfo);
+		if (list == null)
+		{
+			return;
+		}
+
+		Collection<QuestWidget> quests = questSet.get(questContainer);
+
+		if (quests != null)
+		{
+			// Check to make sure the list hasn't been rebuild since we were last her
+			// Do this by making sure the list's dynamic children are the same as when we last saw them
+			if (quests.stream().noneMatch(w ->
+			{
+				Widget codeWidget = w.getQuest();
+				if (codeWidget == null)
+				{
+					return false;
+				}
+				return list.getChild(codeWidget.getIndex()) == codeWidget;
+			}))
+			{
+				quests = null;
+			}
+		}
+
+		if (quests == null)
+		{
+			// Find all of the widgets that we care about, grouping by their Y value
```

Comment:
this doesn't need to do this because there is only 1 widget per quest.
```
quests = list.getDynamicChildren().stream()
.sorted(Comparator.comparing(Widget::getRelativeY))
.map(w -> new QuestWidget(w, Text.removeTags(w.getText()).toLowerCase()))
.collect(Collectors.toList())
```